app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx

```diff
@@ -122,11 +122,24 @@ interface AuthContextType {
   readonly changePassword: (oldPassword: string, newPassword: string) => Promise<boolean>
   readonly resetPassword: (email: string, code: string, password: string) => Promise<boolean>
   readonly signOut: () => Promise<boolean>
+  readonly restoreUser: (backend: Backend) => Promise<boolean>
   /** Session containing the currently authenticated user's authentication information.
    *
    * If the user has not signed in, the session will be `null`. */
   readonly session: UserSession | null
   readonly setUser: React.Dispatch<React.SetStateAction<backendModule.User>>
+  /**
+   * Returns `true` if the user is marked for deletion.
```

Comment:
avoid using singular verbs (`Returns` -> `Return`):
don't do: `[this] returns a b c`
prefer: `[what this does is] return a b c`